crates/runtime/src/embeddings/execution_plan.rs

```diff
@@ -365,7 +374,11 @@ async fn get_vectors_with_chunker(
         values.append_slice(&inner); // I believe this is a clone under the hood.
 
         // Get the length of the last chunk
-        let last_chunk_length = embedded_data.as_slice()[curr + chunks_in_row - 1].len();
+        let last_chunk_length = embedded_data
```

Comment:
I'd like to see some tests to cover this logic. I can see that the `.get` method on a slice will return `None` if it goes out of bounds: https://doc.rust-lang.org/std/primitive.slice.html#method.get so it should not cause a panic, but would return 0. Is it valid for `last_chunk_length` to be 0 in this context?

Reply:
Is okay for it to be zero, see [above](https://github.com/spiceai/spiceai/pull/2940#discussion_r1786948207).